src/game/client/components/mapimages.cpp

```diff
@@ -158,131 +158,133 @@ void CMapImages::LoadBackground(class CLayers *pLayers, class IMap *pMap)
 
-		char aPath[64];
-		str_format(aPath, sizeof(aPath), "%s/%s.png", m_aEntitiesPath, gs_apModEntitiesNames[EntitiesModType]);
-
 		int TextureLoadFlag = 0;
 		if(Graphics()->HasTextureArraysSupport())
 			TextureLoadFlag = (Graphics()->Uses2DTextureArrays() ? IGraphics::TEXLOAD_TO_2D_ARRAY_TEXTURE : IGraphics::TEXLOAD_TO_3D_TEXTURE) | IGraphics::TEXLOAD_NO_2D_TEXTURE;
 
 		CImageInfo ImgInfo;
-		bool ImagePNGLoaded = false;
-		if(Graphics()->LoadPNG(&ImgInfo, aPath, IStorage::TYPE_ALL))
-			ImagePNGLoaded = true;
-		else
+		char aPath[IO_MAX_PATH_LENGTH];
+		str_format(aPath, sizeof(aPath), "%s/%s.png", m_aEntitiesPath, gs_apModEntitiesNames[EntitiesModType]);
+		Graphics()->LoadPNG(&ImgInfo, aPath, IStorage::TYPE_ALL);
+
+		// try as single ddnet replacement
+		if(ImgInfo.m_pData == nullptr && EntitiesModType == MAP_IMAGE_MOD_TYPE_DDNET)
 		{
-			bool TryDefault = true;
-			// try as single ddnet replacement
-			if(EntitiesModType == MAP_IMAGE_MOD_TYPE_DDNET)
-			{
-				str_format(aPath, sizeof(aPath), "%s.png", m_aEntitiesPath);
-				if(Graphics()->LoadPNG(&ImgInfo, aPath, IStorage::TYPE_ALL))
-				{
-					ImagePNGLoaded = true;
-					TryDefault = false;
-				}
-			}
+			str_format(aPath, sizeof(aPath), "%s.png", m_aEntitiesPath);
+			Graphics()->LoadPNG(&ImgInfo, aPath, IStorage::TYPE_ALL);
+		}
 
-			if(!ImagePNGLoaded && TryDefault)
-			{
-				// try default
-				str_format(aPath, sizeof(aPath), "editor/entities_clear/%s.png", gs_apModEntitiesNames[EntitiesModType]);
-				if(Graphics()->LoadPNG(&ImgInfo, aPath, IStorage::TYPE_ALL))
-				{
-					ImagePNGLoaded = true;
-				}
-			}
+		// try default
+		if(ImgInfo.m_pData == nullptr)
+		{
+			str_format(aPath, sizeof(aPath), "editor/entities_clear/%s.png", gs_apModEntitiesNames[EntitiesModType]);
+			Graphics()->LoadPNG(&ImgInfo, aPath, IStorage::TYPE_ALL);
 		}
 
-		if(ImagePNGLoaded && ImgInfo.m_Width > 0 && ImgInfo.m_Height > 0)
+		if(ImgInfo.m_pData != nullptr)
 		{
 			const size_t PixelSize = ImgInfo.PixelSize();
 			const size_t BuildImageSize = (size_t)ImgInfo.m_Width * ImgInfo.m_Height * PixelSize;
-
-			uint8_t *pTmpImgData = ImgInfo.m_pData;
-			uint8_t *pBuildImgData = (uint8_t *)malloc(BuildImageSize);
+			uint8_t *pBuildImgData = static_cast<uint8_t *>(malloc(BuildImageSize));
 
 			// build game layer
-			for(int n = 0; n < MAP_IMAGE_ENTITY_LAYER_TYPE_COUNT; ++n)
+			for(int LayerType = 0; LayerType < MAP_IMAGE_ENTITY_LAYER_TYPE_COUNT; ++LayerType)
 			{
-				dbg_assert(!m_aaEntitiesTextures[(EntitiesModType * 2) + (int)EntitiesAreMasked][n].IsValid(), "entities texture already loaded when it should not be");
+				dbg_assert(!m_aaEntitiesTextures[(EntitiesModType * 2) + (int)EntitiesAreMasked][LayerType].IsValid(), "entities texture already loaded when it should not be");
 
 				// set everything transparent
 				mem_zero(pBuildImgData, BuildImageSize);
 
-				for(int i = 0; i < 256; ++i)
+				for(int i = TILE_AIR; i < 256; ++i)
```

Comment:
```suggestion
				for(int i = 0; i < 256; ++i)
```
The change made the code unclearer to me. It's supposed to iterate over all possible tile values, no?

Reply:
Done.